test/jit/test_graph_rewrite_passes.py

```diff
@@ -4,58 +4,131 @@
+        self.check_single_replacement("aten::matmul", "aten::linear", torch._C._jit_pass_fuse_linear, model1)
+        self.check_op_presence([], check_not, torch._C._jit_pass_fuse_linear, model1)
+        model1(x1)  # make sure it runs
+
+        model2 = torch.jit.trace(FunctionalLinear(w1, None), [x1])
+        self.check_single_replacement("aten::matmul", "aten::linear", torch._C._jit_pass_fuse_linear, model2)
+        self.check_op_presence([], check_not, torch._C._jit_pass_fuse_linear, model2)
+        model2(x1)  # make sure it runs
 
         # check matmuls are not fused
-        class Matmul(torch.nn.Module):
-            def __init__(self, weight):
-                super(Matmul, self).__init__()
-                self.weight = weight
-
-            def forward(self, x):
-                return torch.matmul(x, self.weight)
-
-        x = torch.rand(5, 6, 5)
-        w = torch.rand(5, 5, 100)
-        model = torch.jit.trace(Matmul(w), [x])
-        torch._C._jit_pass_fuse_linear(model.graph)
-        # check 3d matmul is not fused
-        FileCheck().check("aten::matmul").run(model.graph)
-        FileCheck().check_not("aten::linear").run(model.graph)
-        # make sure it runs
-        model(x)
+        x3 = torch.rand(5, 6, 5)
+        w3 = torch.rand(5, 5, 100)
+        model3 = torch.jit.trace(FunctionalMatmul(w3), [x3])
+        check_not3 = ["aten::linear"]
+        self.check_single_replacement("aten::matmul", "aten::matmul", torch._C._jit_pass_fuse_linear, model3)
+        self.check_op_presence([], check_not3, torch._C._jit_pass_fuse_linear, model3)
+        model3(x3)  # make sure it runs
+
+    def test_vulkan_insert_pre_packed_ops(self):
+        x1 = torch.rand(3)
+        w1 = torch.rand(5, 3)
+        b1 = torch.rand(5)
+        model1 = torch.jit.trace(FunctionalLinear(w1, b1), [x1])
+        check_not1 = ["aten::matmul", "aten::add_", "aten::t"]
+        self.check_single_replacement(
+            "aten::matmul",
+            "vulkan_prepack::linear_run",
+            torch._C._jit_pass_vulkan_insert_prepacked_ops,
+            model1
+        )
+        self.check_op_presence([], check_not1, torch._C._jit_pass_vulkan_insert_prepacked_ops, model1)
+        model1(x1)  # make sure it runs
+
+        conv2d_in_channels = 3
+        conv2d_out_channels = 4
+        conv2d_kernel = 2
+        conv2d_weight = torch.rand(
+            conv2d_out_channels,
+            conv2d_in_channels,
+            conv2d_kernel,
+            conv2d_kernel,
+        )
+        conv2d_bias = torch.rand(conv2d_out_channels)
+        x2_shape = (3, 2, 5)
+        x2 = torch.rand(x2_shape)
+        model2 = torch.jit.trace(FunctionalConv2d(conv2d_weight, conv2d_bias), [x2])
+        check_not2 = ["aten::_convolution"]
+        check_yes2 = ["vulkan_prepack::conv2d_clamp_run", "vulkan_prepack::conv2d_clamp_prepack"]
+        self.check_single_replacement(
+            "aten::_convolution",
+            "prim::Constant",
```

Comment:
I suppose `prim::Constant` also maps to `aten::_convolution` so this is ok but I would have imagined you would check with `vulkan_prepack::..`?

Reply:
This is the issue I address in my comment below where I ask for help.